uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Mail/MessageTest.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_NotificationReceiver as NotificationReceiver;
+use  CRM_HRLeaveAndAbsences_Mail_Message as Message;
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_HRLeaveAndAbsences_Factory_RequestNotificationTemplate as RequestNotificationTemplateFactory;
+
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Mail_MessageTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_Mail_MessageTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_LeaveRequestHelpersTrait;
+  use CRM_HRLeaveAndAbsences_LeaveManagerHelpersTrait;
+  use CRM_HRLeaveAndAbsences_MessageHelpersTrait;
+
+
+  private $leaveRequestTemplateFactory;
+
+  private $leaveContact;
+
+  public function setUp() {
+    CRM_Core_DAO::executeQuery('SET foreign_key_checks = 0;');
+    $this->leaveRequestTemplateFactory = new RequestNotificationTemplateFactory();
+
+    $this->leaveContact = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Staff1', 'last_name' => 'Staff1'], 'staffmember@dummysite.com'
+    );
+
+    $this->leaveRequestStatuses = $this->getLeaveRequestStatuses();
+    $this->leaveRequestDayTypes = $this->getLeaveRequestDayTypes();
+  }
+
+  public function testGetRecipientEmailsReturnsCorrectlyWhenLeaveContactHasLeaveApprover() {
+    $manager1 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Manager1', 'last_name' => 'Manager1'], 'manager1@dummysite.com'
+    );
+    $manager2 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Manager2', 'last_name' => 'Manager2'], 'manager2@dummysite.com'
+    );
+    $manager3 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Manager3', 'last_name' => 'Manager3'], 'manager3@dummysite.com'
+    );
+
+    $this->setLeaveApproverRelationshipTypes(['has Leaves Approved By']);
+
+    // Set manager1 and manager2 only to be leave aprovers for the leave contact
+    $this->setContactAsLeaveApproverOf($manager1, $this->leaveContact, null, null, true, 'has Leaves Approved By');
+    $this->setContactAsLeaveApproverOf($manager2, $this->leaveContact, null, null, true, 'has Leaves Approved By');
+
+    $leaveRequest = LeaveRequest::create([
```

Comment:
Please use the LeaveRequest Fabricator here and make this a bit shorter by relying on its default params